src/types/routeMethods.ts

```diff
@@ -14,9 +14,10 @@ export type RouteMethodOptions = {
   skipRouting: boolean,
 }
 
-export type RouteMethodResponse = PromiseLike<{
+export type RouteMethodResponse = {
   url: string,
-}>
+  then: PromiseLike<Omit<RouteMethodResponse, 'then'>>['then'],
+}
```

Comment:
There are a few ways to write this but this is what I settled on. The other options mean we would need to define a type for what a route method returns minus the `then`. Something like this
```typescript
type RouteMethodResponse = { url: string }
type RouteMethodPromiseLike = RouteMethodResponse & PromiseLike<RouteMethodResponse>
```
And I didn't like having two types because one is always poorly named.

Reply:
would this be the same?

```ts
type ThenableResponse<T> = T & {
  then: PromiseLike<T>,
}

export type RouteMethodResponse = ThenableResponse<{
  url: string,
}>
```

I don't feel strongly, but I do think something like above is simpler

Reply:
Close. It would be T & PromiseLike<T>. but then the return type for a route method would be Thenable rather than RouteMethodResponse. I don't feel strongly but named responses are nice over general types

Reply:
After thinking about it more I think having a `Thenable` type would be nice. We'll probably use it for other types as well.

Reply:
Updated